documentation/writing-policies-configmap-reference.md

```diff
@@ -10,41 +10,84 @@ The Configmap Reference allows the reference of configmap values inside kyverno
 
 To refer Configmap inside any Rule provide the context inside each rule defining the list of configmaps which will be referenced in that Rule.
 
-```
+````yaml
   rules:
-    - name: add-sidecar-pod
+    - name: example-configmap-lookup
       # added context to define the configmap information which will be referred 
       context:
       # unique name to identify configmap
-      - name: mycmapRef
+      - name: dictionary
         configMap: 
           # configmap name - name of the configmap which will be referred
           name: mycmap
           # configmap namepsace - namespace of the configmap which will be referred
-```
+          namespace: test
+````
 
 Referenced Configmap Definition
 
-```
+````yaml
 apiVersion: v1
 data:
-  env: production, sandbox, staging
+  env: production
 kind: ConfigMap
 metadata:
   name: mycmap
-```
+````
 
 # Referring Value
 
 The configmaps that are defined inside rule context can be referred using the unique name that is used to identify configmap inside context.
 
-We can refer it's value using a JMESPATH
+We can refer it's value using a JMESPATH `{{<name>.<data>.<key>}}`. 
 
-`{{<name>.<data>.<key>}}`
+For the above context we can refer it's value by `{{dictionary.data.env}}`, which will be substitued with `production` during policy application.
 
-So for the above context we can refer it's value using
+# Deal with Array of Values
 
-`{{mycmapRef.data.env}}`
+The substitute variable can be an array of values. It allows the JSON format when defining it in the configMap.
```

Comment:
The ConfigMap value can be an array of string values in JSON format. Kyverno will parse the JSON string to a list of strings, so set operations like `In` and `NotIn` can then be applied.